### PR TITLE
Increase test threshold for YOLOv3 on OCL FP16

### PR DESCRIPTION
**Pull request by dkurt**
<cut/>

```
force_builders=Custom,Custom Win,Custom Mac
build_image:Custom=ubuntu-openvino-2020.1.0:16.04
#build_image:Custom=ubuntu-openvino-2019r3.0:16.04
build_image:Custom Win=openvino-2020.2.0
build_image:Custom Mac=openvino-2020.2.0

test_modules:Custom=dnn,python2,python3,java
test_modules:Custom Win=dnn,python2,python3,java
test_modules:Custom Mac=dnn,python2,python3,java

buildworker:Custom=linux-1
# disabled due high memory usage: test_opencl:Custom=ON
test_opencl:Custom=ON
test_bigdata:Custom=1
test_filter:Custom=*YOLO*
```

### CODE REVIEW

**Comment by alalek** [comment]
@dkurt Could you please take a look on OpenCL failures from Custom builder?

**Comment by dkurt** [comment]
@alalek, thanks! Added skip for `DLIE/OCL` and batch size > 1. `DLIE/OCL_FP16` works fine

File: modules/dnn/test/test_darknet_importer.cpp [hunk 1]
```diff
@@ -487,7 +487,8 @@ TEST_P(Test_Darknet_nets, YOLOv3)
 #if defined(INF_ENGINE_RELEASE)
     if (backend == DNN_BACKEND_INFERENCE_ENGINE_NN_BUILDER_2019)
     {
-        if (INF_ENGINE_VER_MAJOR_LE(2018050000) && target == DNN_TARGET_OPENCL)
+        if ((INF_ENGINE_VER_MAJOR_LE(2018050000) || INF_ENGINE_VER_MAJOR_GE(2020020000)) &&
+            target == DNN_TARGET_OPENCL)
             applyTestTag(CV_TEST_TAG_DNN_SKIP_IE_OPENCL, CV_TEST_TAG_DNN_SKIP_IE_VERSION);
         else if (INF_ENGINE_VER_MAJOR_EQ(2019020000))
```
Comment (alalek) [inline modules/dnn/test/test_darknet_importer.cpp:493]: > INF_ENGINE_VER_MAJOR_GE(2020020000)
> else if (INF_ENGINE_VER_MAJOR_EQ(2019020000))

This code path should be updated too.

Also there are failed tests with OV 2019R3 and 2020.1

Comment (dkurt) [reply modules/dnn/test/test_darknet_importer.cpp:493]: Seems to me we can skip `DNN_TARGET_OPENCL` for all the versions and `DNN_TARGET_OPENCL_FP16` for <= 2020.1

Comment (alalek) [reply modules/dnn/test/test_darknet_importer.cpp]: @dkurt Please check code compilation.

Comment (dkurt) [reply modules/dnn/test/test_darknet_importer.cpp]: Sorry, fixed that!

**Pull request merged**